### PR TITLE
Made the Armory Gun Safe a craftable object

### PR DESCRIPTION
**Pull request by TyAshley**
## About the PR

Changed Gun Safes to be craftable with 10 steel, 5 LV cables, and 10 plasteel. Takes longer to construct than regular closets/lockers.

## Why / Balance

If something breaks a safe or if new weapons are unlocked in science, it is impossible for the Warden/HoS to create a new safe to hold these weapons.

## Technical details

YML only changes. 

Create entity "GunSafeBase" that has no default access, and can be constructed with 10 steel, 5 LV cable, and 10 plasteel. It has a price of 335 spesos (280 for 10 plasteel, 50 for 10 steel, and 5 for the 5 LV cables). 

Create entity "GunSafeBaseSecure" with "GunSafeBase" as a parent. This has an access level of "Armory"

Changed filled gun safes to inherit from "GunSafeBaseSecure," so that they still spawn requiring Armory access 

## Media

image
image
image
image
image
image

image
image
image
image

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Renamed entity "GunSafe" to "GunSafeBase" that has no default access, and can be constructed, with a cost of 335 spesos (280 for 10 plasteel, 50 for 10 steel, and 5 for the 5 LV cables). 

Create entity "GunSafeBaseSecure" with "GunSafeBase" as a parent. This has access level of "Armory"

Changed filled gun safes to inherit from "GunSafeBaseSecure" vs "GunSafe"

**Changelog**

:cl: Allen
- add: Made Gun Safes a craftable object (10 steel, 5 lv cable, 10 plasteel)

### CODE REVIEW

**Comment by lzk228** [comment]
it should start without accesses, so ce or hop has to configure it later

**Comment by TyAshley** [comment]
I do see the test failing (and agree it is a valid test failure), and am taking a look at the spesos difference between materials and the safe itself. I'll also need to double-check the process for starting a constructed object with no accesses set.


> > it should start without accesses, so ce or hop has to configure it later
> 
> I agree. It is weird if you can build it, but are unable to open it. Also the test fail is real. If you deconstruct it the materials are worth more spesos than the safe itself.

**Comment by lzk228** [comment]
> I'll also need to double-check the process for starting a constructed object with no accesses set.

yeah, you just need to create gun safe entity without accesses and use it for construction
then entity with accesses should inherit from the first one
and all filled versions should inherit from the one with accesses

**Comment by TyAshley** [comment]
Corrected the cost to match the cost of materials (335 total). I also updated the constructed gun safe to start with no access levels, and need to be adjusted with an access configurator. The map-spawn filled gun safes continue to have an armory access level.

File: Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml [hunk 1]
```diff
@@ -350,7 +350,7 @@
     access: [["Security"]]
 
 - type: entity
-  id: GunSafe
+  id: GunSafeBase
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml:353]: better to not change the id
it's fine with the current

File: Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml [hunk 2]
```diff
@@ -359,6 +359,19 @@
...
+    node: done
+    containers:
+    - entity_storage
+  - type: StaticPrice
+    price: 335
+
+- type: entity
+  id: GunSafeBaseSecure
+  parent: GunSafeBase
+  components:
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml:374]: add a suffix `Armory` or something, so in spawn menu you could see the difference

**Comment by slarticodefast** [comment]
Are you coming back to this?
If yes, please address the review and the test fail.

**Comment by slarticodefast** [comment]
If you are coming back to this, leave a comment and I will re-open the PR.

**Pull request closed without merging**